spec/acceptance/define_versionlock_spec.rb

```diff
@@ -32,11 +32,12 @@
       ensure  => present,
     }
     EOS
-    # Run it twice and test for idempotency
+    # Run it a first time to apply the changes
     apply_manifest(pp, catch_failures: true)
-    apply_manifest(pp, catch_changes:  true)
     # Check the cache is really empty.
     # all repos will have 0 packages.
     shell('yum -C repolist -d0 | grep -v "repo id"  | awk "{print $NF}" FS=  | grep -v 0', acceptable_exit_codes: [1])
+    # Run it a second time and test for idempotency
```

Comment:
This needs be be right after the first to ensure it's really idempotent. The shell command in between can change things.

Reply:
I can't recall the reason for this, the commit message being "Unbreak CI" and indeed when I remove 309aed39650f78d89b04389bed4a007bc6aa76f8, the CI breaks :).

Yet, I am unsure about what this shell command is supposed to check:

1. According to the comment above, how are we supposed to have no remote package known after checking for available updates?
2. The command itself is suspect: `$NF` will be expanded to an empty string before running awk; i don't think awk will do anything with the `FS=` argument; `grep -v 0` may better match `^0$`.

Can you please clarify what is really expected here?

Reply:
Using `yum -C` in c8ce7095fa20cda0ecea9dbccfe1f19f36e9b39f seems to fix the test suite… Yet it's still black magic / cargo cult for me :scream: